File: android/app/src/main/java/com/matthiascadet/holyday/ui/theme/AppConstants.kt
package com.matthiascadet.holyday.ui.theme

import android.content.ActivityNotFoundException
import android.content.Context
import android.content.Intent
import android.net.Uri

/** Équivalent de `AppLinks` (HolyDay/Theme/AppConstants.swift). */
object AppLinks {
    const val PRIVACY_POLICY = "https://holyday-landing.vercel.app/privacy.html"
    const val TERMS_OF_SERVICE = "https://holyday-landing.vercel.app/terms.html"
    const val PLAY_STORE = "https://play.google.com/store/apps/details?id=com.matthiascadet.holyday"
    const val PLAY_STORE_MARKET = "market://details?id=com.matthiascadet.holyday"

    /** Ouvre le formulaire de notation Play Store avec un repli navigateur si Play Store manque. */
    fun openReview(context: Context) {
        try {
            context.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(PLAY_STORE_MARKET)))
        } catch (_: ActivityNotFoundException) {
            context.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(PLAY_STORE)))
        }
    }
}

/**
 * Équivalent de `RevenueCatConfig` (HolyDay/Theme/AppConstants.swift). La clé API Android est
 * distincte de la clé iOS — à créer côté dashboard RevenueCat pour l'app Android, voir le
 * rapport de publication Play Store.
 */
object RevenueCatConfig {
    const val API_KEY = "goog_REPLACE_WITH_ANDROID_PUBLIC_API_KEY"
    const val ENTITLEMENT_ID = "ia_lifetime"
    const val AI_ENTITLEMENT_ID = "ia_feature"
    const val OFFERING_ID = "tips"
    const val AI_OFFERING_ID = "default"
}

File: ios/HolyDay/Theme/AppConstants.swift
//
//  AppConstants.swift
//  Holyday
//
//  Created by Matthias Cadet on 13/05/2026.
//

import Foundation

// TODO: Replace these URLs with your actual policy pages before App Store submission
enum AppLinks {
  static let privacyPolicy = makeURL("https://holyday-landing.vercel.app/privacy.html")
  static let termsOfService = makeURL("https://holyday-landing.vercel.app/terms.html")
  static let appStore = makeURL("https://apps.apple.com/us/app/holyday/id6774578624")
  /// Ouvre directement le composeur d'avis de l'App Store. À préférer à `requestReview` dès que
  /// c'est l'utilisateur qui demande à noter : la fenêtre système est bridée par Apple (3 fois par
  /// an au plus, et pas du tout si l'utilisateur a désactivé les avis in-app), donc un bouton qui
  /// s'appuie dessus peut ne produire aucune réaction visible.
  static let writeReview = makeURL(
    "https://apps.apple.com/app/id6774578624?action=write-review")

  private static func makeURL(_ string: String) -> URL {
    guard let url = URL(string: string) else {
      preconditionFailure("URL statique invalide : \(string)")
    }
    return url
  }
}

enum RevenueCatConfig {
  // Clé SDK *publique* RevenueCat (préfixe `appl_`) : conçue pour être embarquée dans le binaire
  // et extractible de toute app publiée — ce n'est pas un secret. La garder en clair ici est
  // conforme aux recommandations RevenueCat.
  static let apiKey = "appl_UlQUPWYbfJUrWXoDkEkNxuQHZkY"
  static let entitlementId = "ia_lifetime"
  static let aiEntitlementId = "ia_feature"
  static let offeringId = "tips"
  static let aiOfferingId = "default"
}
